torch_frame/gbdt/tuned_lightgbm.py

```diff
@@ -0,0 +1,202 @@
+            raise ValueError("The input TensorFrame object is empty.")
+
+        feat = torch.cat(feats, dim=-1)
+
+        return feat.numpy(), y.numpy()
+
+    def _predict_helper(
+        self,
+        model: Any,
+        x: DataFrame,
+    ) -> np.ndarray:
+        r"""A helper function that applies the lightgbm model on DataFrame
+        :obj:`x`.
+
+        Args:
+            model (lightgbm.Booster): The lightgbm model.
+            x (DataFrame): The input `DataFrame`.
+
+        Returns:
+            pred (np.ndarray): The prediction output.
+        """
+        pred = model.predict(x)
+        if self.task_type == TaskType.MULTICLASS_CLASSIFICATION:
+            pred = pred.argmax(axis=1)
+
+        return pred
+
+    def objective(
+        self,
+        trial: Any,
+        tf_train: TensorFrame,
+        tf_val: TensorFrame,
+        num_boost_round: int,
+    ):
+        r"""Objective function to be optimized.
+
+        Args:
+            trial (optuna.trial.Trial): Optuna trial object.
+            tf_train (TensorFrame): Train data.
+            tf_val (TensorFrame): Validation data.
+            num_boost_round (int): Number of boosting round.
+
+        Returns:
+            float: Best objective value. Mean absolute error for
+            regression task and accuracy for classification task.
+        """
+        import lightgbm
+
+        self.params = {
+            "verbosity":
+            -1,
+            "bagging_freq":
+            1,
+            "max_depth":
+            trial.suggest_int("max_depth", 3, 11),
+            "learning_rate":
+            trial.suggest_float("learning_rate", 1e-3, 0.1, log=True),
+            "num_leaves":
+            trial.suggest_int("num_leaves", 2, 2**10),
+            "subsample":
+            trial.suggest_float("subsample", 0.05, 1.0),
+            "colsample_bytree":
+            trial.suggest_float("colsample_bytree", 0.05, 1.0),
+            'reg_alpha':
+            trial.suggest_float('reg_alpha', 1e-9, 10.0, log=True),
+            'reg_lambda':
+            trial.suggest_float('reg_lambda', 1e-9, 10.0, log=True),
+            'min_child_samples':
+            trial.suggest_int('min_child_samples', 5, 100),
+            "min_data_in_leaf":
+            trial.suggest_int("min_data_in_leaf", 1, 100),
+        }
+
+        if self.task_type == TaskType.REGRESSION:
+            if self.metric == Metric.RMSE:
+                self.params["objective"] = "regression"
+                self.params["metric"] = "rmse"
+            elif self.metric == Metric.MAE:
+                self.params["objective"] = "regression_l1"
+                self.params["metric"] = "mae"
+        elif self.task_type == TaskType.BINARY_CLASSIFICATION:
+            self.params["objective"] = "binary"
+            if self.metric == Metric.ROCAUC:
+                self.params["metric"] = "auc"
+            elif self.metric == Metric.ACCURACY:
+                self.params["metric"] = "binary_logloss"
```

Comment:
Doesn't LightGBM offer accuracy?

Reply:
https://lightgbm.readthedocs.io/en/latest/Parameters.html#metric

Reply:
I think binary error is what you should use. It’s equivalent to accuracy